packages/aws-cdk-lib/aws-route53/lib/hosted-zone.ts

```diff
@@ -132,11 +139,6 @@ export class HostedZone extends Resource implements IHostedZone {
       props: query,
     }).value;
 
-    // CDK handles the '.' at the end, so remove it here
-    if (response.Name.endsWith('.')) {
```

Comment:
I think we need to keep this here. It looks like we are explicitly keeping the `Name` value within cdk without the `.` and I don't think we should change this behavior.